indexer/src/main.rs

```diff
@@ -156,16 +179,21 @@ async fn main() -> anyhow::Result<()> {
     let receipts_cache: receipts_cache::ReceiptsCache =
         std::sync::Arc::new(Mutex::new(SizedCache::with_size(100_000)));
 
-    let config: near_lake_framework::LakeConfig = opts.to_lake_config().await;
-    let (_, stream) = near_lake_framework::streamer(config);
-
-    tokio::spawn(metrics::init_server(opts.port).expect("Failed to start metrics server"));
-
     tracing::info!(
         target: INDEXER_FOR_EXPLORER,
         "Starting Indexer for Explorer (lake)...",
     );
 
+    tokio::spawn(metrics::init_server(opts.port).expect("Failed to start metrics server"));
+
+    if opts.start_options() == &StartOptions::FromGenesis {
+        let genesis_file_path = download_genesis_file(&opts).await?;
+        adapters::genesis::store_genesis_records(&pool, genesis_file_path).await?;
+    }
+
+    let config: near_lake_framework::LakeConfig = opts.to_lake_config().await;
+    let (_, stream) = near_lake_framework::streamer(config);
```

Comment:
While you are here could you replace the `_` with the `sender` and handle the possible errors like it is suggested in the Lake Framework README, please?

ref: https://github.com/near/near-lake-framework-rs/blob/main/README.md?plain=1#L36-L53

Reply:
Done